scripts/ast-diff

```diff
@@ -0,0 +1,23 @@
+#! /usr/bin/env bash
+
+function filter
+{
+    sed 's/ \{0,\}\[@.:[^]]*\]//g' | \
+    sed 's/ \{0,\}([^)]*:[0-9-]*:[0-9-]*)//g' |
+    sed 's/ \{0,\}%[0-9-]\{1,\}//g'
+}
+
+if [ $# != 2 ]; then
+    echo "usage: $(basename $0) <ast1> <ast2>"
+    exit 1
+fi
+
+tmp1=/tmp/$(basename $0).$$.1
+tmp2=/tmp/$(basename $0).$$.2
+
+trap "rm -f ${tmp1} ${tmp2}" EXIT
+
+cat $1 | filter >${tmp1}
+cat $2 | filter >${tmp2}
+
+diff -u ${tmp1} ${tmp2}
```

Comment:
Just FTR, when I tried such a script before I always ran into issues with the `diff` binary behaving differently on different platforms; I then instead implemented the script in Python and used [`difflib`](https://docs.python.org/3/library/difflib.html) (e.g., [`difflib.unifed_diff`](https://docs.python.org/3/library/difflib.html#difflib.unified_diff) here).

This is not an issue now since we do not use this script to compute baselines, but we might need to move this whole script to Python at that point.

Reply:
Ack. This script is meant for manual debugging, not baselining, so something we could change later if we wanted to do that.